src/main/kotlin/org/jitsi/nlj/transform/node/incoming/VideoParser.kt

```diff
@@ -47,28 +46,23 @@ class VideoParser(
     // does this packet represent the start of a frame?
     // does this packet represent the end of a frame?
     override fun transform(packetInfo: PacketInfo): PacketInfo? {
-        val rtpPacket = packetInfo.packetAs<RtpPacket>()
-        return streamInformationStore.rtpPayloadTypes[rtpPacket.payloadType.toByte()]?.let { payloadType ->
-            val videoRtpPacket = when (payloadType) {
-                is Vp8PayloadType -> {
-                    val vp8Packet = rtpPacket.toOtherType(::Vp8Packet)
-                    findRtpEncodingDesc(vp8Packet)?.let {
-                        vp8Packet.qualityIndex = it.index
-                        vp8Packet
-                    } ?: run {
-                        numVp8PacketsDroppedNoEncoding.incrementAndGet()
-                        null
-                    }
-                }
-                else -> rtpPacket
-            } ?: return null
-            packetInfo.packet = videoRtpPacket
-            packetInfo.resetPayloadVerification()
-            packetInfo
-        } ?: run {
-            logger.error("Unrecognized video payload type ${rtpPacket.payloadType}, cannot parse video information")
-            null
+        val videoPacket = packetInfo.packetAs<VideoRtpPacket>()
+        val payloadType = streamInformationStore.rtpPayloadTypes[videoPacket.payloadType.toByte()] ?: run {
+            logger.error("Unrecognized video payload type ${videoPacket.payloadType}, cannot parse video information")
+            return null
         }
+        val encodingDesc = findRtpEncodingDesc(videoPacket) ?: run {
+            logger.warn("Unable to find encoding matching packet! encodings=$tracks, packet=$videoPacket")
```

Comment:
Hm.  In the previous flow, we wouldn't even call `findRtpEncodingDesc` for non-VP8 packets.  Does it do anything useful for them?

Reply:
Other codecs should still have encoding descriptions, so @bgrozev brought up dropping for all in the same place, which seemed reasonable.